clvm/CLVMObject.py

```diff
@@ -1,5 +1,23 @@
+from __future__ import annotations
+
 import typing
 
+from typing_extensions import Protocol
+
+
+class CLVMObjectLike(Protocol):
```

Comment:
Check out https://github.com/Chia-Network/clvm_rs/blob/main/wheel/python/clvm_rs/clvm_storage.py which is the future of `clvm`. It has similar ideas with slightly different names.